subsys/bluetooth/audio/unicast_client.c

```diff
@@ -1208,9 +1224,10 @@ static void unicast_client_ep_set_cp(struct bt_conn *conn, uint16_t handle)
 		cp_subscribe[index].value_handle = handle;
```

Comment:
I don't see where `cp_subscribe[?].value_handle` is cleared. Perhaps it isn't.

I think it's harmless to always subscribe to CCC here. If `cp_subscribe[?].value_handle` is re-used from one connection to another then it's essential.

Either way `params->func(conn, NULL, NULL, params)` needs to be called eventually so if we keep this `if ()` there should be an `else` clause?

Reply:
Assuming that `bt_audio_discover` is always called for a new connection (which it should, given that the BAP implementation does not really support bonded information yet), then it does not need to be cleared.

Ideally we probably should clear it, and all the other connection-based values, on disconnect. However adding that is a separate issue, and something I'd rather do in a separate PR (ideally after https://github.com/zephyrproject-rtos/zephyr/pull/54363/commits/f35c89d855cca74aaaeff222708469ec76de7d2d has been merged, to make it easier).

Reply:
my worry is what happens if a client connects, discovers, disconnects, connects to a different server, discovers?

Do connection indices get re-used? I think they might. Will cp_subscribe[0].value_handle be zero? Probably not.

Reply:
> Do connection indices get re-used? I think they might

Correct

> Will cp_subscribe[0].value_handle be zero? Probably not.

Correct, but it will get overwritten with the actual handle. 

I don't disagree that we should clear the data, I only suggest that we postpone that until my other PR has been merged, which would make it significantly easier to handle. The lack of clearing of data here shouldn't affect the issue that this PR is fixing